### ISSUE TITLE
Array slicing not working with redux replay.

### ISSUE THREAD

**Issue by OnurGumus** [elmish/debugger#15]
### Description

I have the following code
```
let viewTime (timer : int array) =
    timer.[0..2] 
        |> Array.map (fun s -> s.ToString("00")) 
        |> String.concat ":"
```
This seems to work fine, but with redux replay I get the following error:

`timer.slice is not a function`

* fable-compiler version: 2.0.6

**Comment by et1975** [comment]
What is `Time`? @MangelMaxime do you think it can survive the roundtrip?

**Comment by OnurGumus** [comment]
@et1975  It is an alias to int array. I have updated the code above.

**Comment by et1975** [comment]
Can you give 2.0.1 release a spin?

**Comment by OnurGumus** [comment]
I think it is still indexing:
`
-> Unable to retrieve package details for 'Fable.Elmish.Debugger'-2.0.1
-> Couldn't get package details for package Fable.Elmish.Debugger 2.0.1 on https://api.nuget.org/v3/index.json.
`

**Comment by OnurGumus** [comment]
@et1975  I am getting this error ```DecoderException {Data0: Array(2)}```
`
F:95 Unable to process monitor command {type: "DISPATCH", payload: {…}, state: "{"HMRCount":0,"UserModel":{"Time":{"0":0,"1":0,"2"…s when you match this text exactly. Good Luck!"}}", id: "1", source: "@devtools-extension"} DecoderException {Data0: Array(2)}
(anonymous) @ F:95
(anonymous) @ VM121:1
(anonymous) @ VM121:1
(anonymous) @ VM121:1
handleMessages @ VM121:1
postMessage (async)
(anonymous) @ content.bundle.js:1
EventImpl.dispatchToListener @ VM143 extensions::event_bindings:403
publicClassPrototype.(anonymous function) @ VM149 extensions::utils:138
EventImpl.dispatch_ @ VM143 extensions::event_bindings:387
EventImpl.dispatch @ VM143 extensions::event_bindings:409
publicClassPrototype.(anonymous function) @ VM149 extensions::utils:138
dispatchOnMessage @ VM150 extensions::messaging:392
`

**Comment by OnurGumus** [comment]
The good news is the other list issue seems to be fixed.

**Comment by MangelMaxime** [comment]
@et1975 I think it should.

You are getting `DecoderException ...` because we are using [`unwrap`](https://github.com/elmish/debugger/blob/master/src/debugger.fs#L86) instead of `fromValue` which capture the `DecoderException` exception and print the correct message.

**Comment by OnurGumus** [comment]
Anything expected from me regarding this issue?

**Comment by et1975** [comment]
@alfonsogarciacaro @MangelMaxime I've been trying to understand what might be the issue here, but I can't. I've been able to use the indexer (slice) on deserialized array, however if I understand the scenario, the problem here happens *during deserialization*, not after.
@OnurGumus are you perhaps loading an exported history into the debugger and simply seeing a model mismatch? Is there a repo we can poke at?

**Comment by MangelMaxime** [comment]
@et1975 Indeed the problem is happening during the deserialization since we are getting a `DecoderException` which is coming from Thoth.Json.

This exception is internal to Thoth.Json but because we use `unwrap` it's not captured correctly. This is why we should use `fromValue` instead of `unwrap` so we can have the "nice error display" and probably have more info on what is the error exactly. 

I am not sure I can take a look at it this week because we have the FableConf but I will add a reminder to look at it next week.

**Comment by OnurGumus** [comment]
@et1975  nope pretty standard usage, not exporting. I will try to give you a repo soon.

**Comment by et1975** [comment]
@MangelMaxime Rising the exception is the expected behavior here, what I don't understand is why an array would not roundtrip. @alfonsogarciacaro, you had it as a WIP PR, does it mean you know something still missing?

**Comment by alfonsogarciacaro** [comment]
@et1975 Sorry, I hadn't tested my previous PR. I did it know and I think #17 should work.

**Comment by et1975** [comment]
Released as 2.0.2.

**Comment by OnurGumus** [comment]
@et1975 I can confirm it works. However, I created another project and it was working fine with 2.0.1 too! Perhaps it was something cached ?  I don't know. I am sorry if I caused an effort for no reason. I am closing the issue as it seems to be fixed.
